### PR TITLE
refactor(modularize): functionalities

### PR DESCRIPTION
**Pull request by sarbbottam**
modularized `inject` function; extracted out most of the logic out of it.

Further refactor needed: wanted to keep the use case as little as possible. Please see the inline comments w.r.t. [moment.js](http://momentjs.com/)

### CODE REVIEW

File: src/inject.js [hunk 1]
```diff
@@ -1,120 +1,184 @@
...
-        const days = ([].slice.call(document.getElementsByClassName("day"))).reverse()
+    let totalContributionsText = ""
+    let longestStreakText = ""
+    let currentStreakText = ""
+    let lastContributionText = ""
+
+    let firstContributionDate = ""
+    let lastContributionDate = ""
+    let longestStreakStartingDate = ""
+    let longestStreakEndingDate = ""
```
Comment (sarbbottam) [inline src/inject.js:111]: renamed variable to better semantics

File: src/inject.js [hunk 2]
```diff
@@ -1,120 +1,184 @@
...
-    let hasNoCurrentStreak = 0
-    let streakFromTo = ""
-    let innerStreak = ""
-    let firstCommitDate = 0
-    let lastCommitDate = 0
-    let longestStreakStartingDate = 0
-    let longestStreakEndingDate = 0
-
-    if (!!document.getElementById("contributions-calendar")) {
-        let couple = document.getElementById("contributions-calendar").previousElementSibling.childNodes
```
Comment (sarbbottam) [inline src/inject.js:18]: cache `document.getElementById("contributions-calendar")`, see below

File: src/inject.js [hunk 3]
```diff
@@ -1,120 +1,184 @@
...
+    let longestStreakText = ""
+    let currentStreakText = ""
+    let lastContributionText = ""
+
+    let firstContributionDate = ""
+    let lastContributionDate = ""
+    let longestStreakStartingDate = ""
+    let longestStreakEndingDate = ""
+
+    const contributionsCalendar = document.getElementById("contributions-calendar")
```
Comment (sarbbottam) [inline src/inject.js:113]: cached `document.getElementById("contributions-calendar")`

File: src/inject.js [hunk 4]
```diff
@@ -1,120 +1,184 @@
...
-    let longestStreakStartingDate = 0
-    let longestStreakEndingDate = 0
-
-    if (!!document.getElementById("contributions-calendar")) {
-        let couple = document.getElementById("contributions-calendar").previousElementSibling.childNodes
-        if (couple.length === 1) {
-            couple[0].textContent = "Contributions"
-        } else {
-            couple[2].textContent = "Contributions"
-        }
```
Comment (sarbbottam) [inline src/inject.js:23]: moved this to `updateContributionsHeading` function

File: src/inject.js [hunk 5]
```diff
@@ -1,120 +1,184 @@
...
+                firstContributionDate = new Date(contributionDate)
+
+                if (!lastContributionDate) {
+                    lastContributionDate = new Date(contributionDate)
+                }
+
+                currentLongestStreak += 1
+            } else {
+                currentLongestStreak = 0
             }
```
Comment (sarbbottam) [inline src/inject.js:134]: grouped all the calculation when `contributionCount` is more than 0

File: src/inject.js [hunk 6]
```diff
@@ -1,120 +1,184 @@
...
-                ++nonContributingDays
+                nonContributingStreak += 1
                 if (stop === 0) {
-                    const start = new Date(day.attributes["data-date"].value)
-                    const end = new Date()
-                    const dayF = start.getUTCDate() + 1
-                    const monthF = monthNames[start.getMonth()]
-                    const dayT = end.getUTCDate()
-                    const monthT = monthNames[end.getMonth()]
-                    streakFromTo = `${dayF} ${monthF} - ${dayT} ${monthT}`
```
Comment (sarbbottam) [inline src/inject.js:46]: moved the computation to `getCurrentStreakText` function

File: src/inject.js [hunk 7]
```diff
@@ -1,120 +1,184 @@
...
-            if (todayValue) {
-                ++evaluatingStreak
-            } else {
-                evaluatingStreak = 0
             }
-            if (evaluatingStreak > longestStreak) {
-                longestStreak = evaluatingStreak
-                longestStreakStartingDate = new Date(day.attributes["data-date"].value)
-                longestStreakEndingDate = new Date(day.attributes["data-date"].value)
-                longestStreakEndingDate.setDate(longestStreakEndingDate.getDate() + longestStreak - 1)
```
Comment (sarbbottam) [inline src/inject.js:78]: this has moved above, next to `if (contributionCount) {` block.

File: src/inject.js [hunk 8]
```diff
@@ -1,120 +1,184 @@
...
                 }
-                if (timeDiff > 30) {
-                    unit = "month"
-                    if (timeDiff > 60) {
-                        unit = "months"
-                    }
-                    timeDiff = (new Date()).getMonth() - lastCommit.getMonth() + (12 * ((new Date()).getFullYear() - lastCommit.getFullYear()))
-                }
-                hasNoCurrentStreak = `Last contributed <time>${timeDiff} ${unit} ago</time>`
-            }
```
Comment (sarbbottam) [inline src/inject.js:68]: extracted out the functionality to `getLastContributionText`

File: src/inject.js [hunk 9]
```diff
@@ -1,120 +1,184 @@
...
-                    ${monthNames[longestStreakStartingDate.getMonth()]} ${longestStreakStartingDate.getUTCDate()} –
-                    ${monthNames[longestStreakEndingDate.getMonth()]} ${longestStreakEndingDate.getUTCDate()}
-                    </span>
-                </div>
-                <div class="contrib-column table-column">
-                    <span class="text-muted">Current streak</span>
-                    <span class="contrib-number">${currentStreak} days</span>
-                    ${currentStreakSkeleton}
-                </div>`
-            let range = document.createRange()
```
Comment (sarbbottam) [inline src/inject.js:114]: extracted out the logic to `getStreakHTML` function.

File: src/inject.js [hunk 10]
```diff
@@ -1,120 +1,184 @@
...
-            const yearTo = lastCommitDate.getFullYear()
-            let currentStreakSkeleton = ""
-            if (hasNoCurrentStreak) {
-                currentStreakSkeleton = `<span class="text-muted">${hasNoCurrentStreak}</span>`
-            } else {
-                currentStreakSkeleton = `<span class="text-muted">${innerStreak}</span>`
+        if (firstContributionDate) {
+            currentStreakText = lastContributionText ? lastContributionText : currentStreakText
+            longestStreakText = getLongestStreakText(longestStreakStartingDate, longestStreakEndingDate)
+            totalContributionsText = getTotalContributionsText(firstContributionDate, lastContributionDate)
```
Comment (sarbbottam) [inline src/inject.js:163]: these are not required to to be associated to a `var`; just for the sake of consistency in the first pass. will be removed in next PR.

**Comment by Naramsim** [comment]
@sarbbottam You are working too much :smile: I'll review all of them it in 3 hours.

**Pull request merged**